test/TestUtils/TestProjectFixture.cs

```diff
@@ -0,0 +1,290 @@
+
+            var buildArgs = new List<string>();
+            if (runtime != null)
+            {
+                buildArgs.Add("--runtime");
+                buildArgs.Add(runtime);
+            }
+
+            if (framework != null)
+            {
+                buildArgs.Add("--framework");
+                buildArgs.Add(framework);
+            }
+
+            if (outputDirectory != null)
+            {
+                buildArgs.Add("-o");
+                buildArgs.Add(outputDirectory);
+            }
+
+            dotnet.Build(buildArgs.ToArray())
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .EnsureSuccessful();
+
+            _testProject.LoadOutputFiles();
+
+            return this;
+        }
+
+        public TestProjectFixture PublishProject(
+            DotNetCli dotnet = null,
+            string runtime = null,
+            string framework = "netcoreapp1.0",
+            string outputDirectory = null)
+        {
+            dotnet = dotnet ?? _sdkDotnet;
+            outputDirectory = outputDirectory ?? _testProject.OutputDirectory;
+            _testProject.OutputDirectory = outputDirectory;
+
+            var publishArgs = new List<string>();
+            if (runtime != null)
+            {
+                publishArgs.Add("--runtime");
+                publishArgs.Add(runtime);
+            }
+
+            if (framework != null)
+            {
+                publishArgs.Add("--framework");
+                publishArgs.Add(framework);
+            }
+
+            if (outputDirectory != null)
+            {
+                publishArgs.Add("-o");
+                publishArgs.Add(outputDirectory);
+            }
+
+            dotnet.Publish(publishArgs.ToArray())
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .EnsureSuccessful();
+
+            _testProject.LoadOutputFiles();
+
+            return this;
+        }
+
+        public TestProjectFixture RestoreProject(DotNetCli dotnet = null)
+        {
+            dotnet = dotnet ?? _sdkDotnet;
+
+            dotnet.Restore()
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .Execute()
+                .EnsureSuccessful();
+
+            return this;
+        }
+
+        public TestProjectFixture EnsureRestored(DotNetCli dotnet = null)
+        {
+            dotnet = dotnet ?? _sdkDotnet;
+
+            if ( ! _testProject.IsRestored())
+            {
+                RestoreProject(dotnet: dotnet);
+            }
+
+            return this;
+        }
+
+        public TestProjectFixture Copy()
+        {
+            return new TestProjectFixture(this);
+        }
+
+        public TestProjectFixture MoveDepsJsonToSubdirectory()
```

Comment:
These methods are only called from tests in the GivenThatICareAboutPortableAppActivation class.

I would suggest just making them `private static` methods in that class.


---
In reply to: [66350557](https://github.com/dotnet/core-setup/pull/97#discussion_r66350557) [](ancestors = 66350557,66349519,66107124)